src/server/src/main/java/io/cassandrareaper/service/RepairScheduleService.java

```diff
@@ -154,15 +154,17 @@ private void registerScheduleMetrics(UUID repairScheduleId) {
   }
 
   private void unregisterScheduleMetrics(UUID repairScheduleId) {
-    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
-    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
-    String metricName = metricName("millisSinceLastRepairForSchedule",
-        repairUnit.getClusterName(),
-        repairUnit.getKeyspaceName(),
-        schedule.getId());
-
-    if (context.metricRegistry.getMetrics().containsKey(metricName)) {
-      context.metricRegistry.remove(metricName);
+    Optional<RepairSchedule> schedule = context.storage.getRepairSchedule(repairScheduleId);
+    if (schedule.isPresent()) {
+      RepairUnit repairUnit = context.storage.getRepairUnit(schedule.get().getRepairUnitId());
+      String metricName = metricName("millisSinceLastRepairForSchedule",
+          repairUnit.getClusterName(),
+          repairUnit.getKeyspaceName(),
+          schedule.get().getId());
+
+      if (context.metricRegistry.getMetrics().containsKey(metricName)) {
+        context.metricRegistry.remove(metricName);
+      }
```

Comment:
In the integration tests, we have several reaper that compete to delete the schedule, which made it so that on reaper could try to remove it after another one removed it. Hence, the optional would be empty and the test would fail.
This change allows avoiding such failures that create flakiness in the tests.

Reply:
This comment brings me to another question, is that race condition really intended? If so, this isn't really safe alternative either? If we get the repairUnit and then another thread removes it, we're now re-removing data from metricRegistry since these two threads are racing to delete the metric.

Reply:
It's not several threads in the same Reaper instance, but several Reaper instances running in different JVMs. The problem here is that the metric won't get cleaned up on other Reaper instances, making that cleanup mostly inefficient in distributed mode 😕 
I think we could improve this by listing the schedules and checking if there are corresponding schedules for the repair unit. This way we don't need to have the schedule in storage for cleanup to happen.
Let me try that.

Reply:
Now that I think of it, unlike during tests a schedule deletion will only be sent to a single instance. This means metrics never get fully cleaned up in a distributed setup.
We'd need to regularly clean them up. I'll open another ticket to handle this.

Reply:
I've created the follow up issue: #1211